File: BarNone/Shared/BarNone.Shared.Analysis/LiftAnalysisPipeline/Core/AnalysisRequest.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Text;

namespace BarNone.Shared.Analysis.LiftAnalysisPipeline.Core
{
    public class AnalysisRequest
    {
        //public AnalysisRequest()
        //{
        //    Requests = new List<RequestEntity>();
        //}

        public List<object> Requests { get; set; }
    }


    public class RequestEntity : IAnalysisRequest
    {
        public ELiftAnalysisType Type { get; set; }
    }
}

File: BarNone/TheBar/src/App/Data/Models/Analysis/AnalysisRequest.ts
export default class AnalysisRequest {
	public requests: RequestEntity[];
}

export enum ELiftAnalysisType {
	Position = 1,
	Velocity = 2,
	Acceleration = 3,
	Angle = 4
}

export class RequestEntity {
	[key: string]: any;

	public type: ELiftAnalysisType;
}